internal/kubelet/client.go

```diff
@@ -238,9 +282,12 @@ func (c *clientImpl) Get(path string) ([]byte, error) {
 	return body, nil
 }
 
-func (c *clientImpl) buildReq(path string) (*http.Request, error) {
-	url := c.baseURL + path
-	req, err := http.NewRequest("GET", url, nil)
+func (c *clientImpl) buildReq(p string) (*http.Request, error) {
+	reqURL, err := url.JoinPath(c.baseURL, p)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("GET", reqURL, nil)
```

Comment:
I understand this is the right change, but it's unrelated to the PR's goal. Feel free to create a separate PR with `[chore]` label (no need for a changelog entry)

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19212

Can this PR be merged so I can rebase on it ? Otherwise I need to modify the kubeclient test in a weird way suffixing the expected path with `//` instead of `/`.

Reply:
Merged, please rebase

Reply:
done